pkg/api/vlabs/types.go

```diff
@@ -37,6 +37,11 @@ type Properties struct {
 	WindowsProfile          *WindowsProfile          `json:"windowsProfile,omitempty"`
 	ServicePrincipalProfile *ServicePrincipalProfile `json:"servicePrincipalProfile,omitempty"`
 	CertificateProfile      *CertificateProfile      `json:"certificateProfile,omitempty"`
+	CustomProfile           *CustomProfile           `json:"customProfile,omitempty"`
+}
+
+type CustomProfile struct {
```

Comment:
Based on our discussion I thought kubernetesagentpool was completely separate from vlabs? this little bit is confusing me